src/main/scala/org/apache/spark/sql/BebeFunctions.scala

```diff
@@ -39,6 +39,23 @@ object BebeFunctions {
       BeginningOfMonth(col.expr)
     }
 
+  def endOfDay(col: Column, timeZoneId: Option[String] = None): Column =
+    withExpr {
+      val startOfNextDay =
+        TruncTimestamp(lit("day").expr, DateAdd(col.expr, lit(1).expr), timeZoneId)
+      // TODO: Is this definition of end of day common? Maybe we shouldn't have an
```

Comment:
This is inspired by the Rails [end_of_day function](https://apidock.com/rails/DateTime/end_of_day).

Not sure if it's standard beyond Rails.

Reply:
Do you want the bebe version to work like Rails in that it's a second before the end of day? Currently I have it at a microsecond.

Reply:
Whatever you think it better.  Trust your judgement more than mine ;)

Reply:
Woudn't it be better to publish the function with two intefaces to prevent te user to pass a Option?
```scala
def endOfDay(col: Column): Column = // equal to pass a None
def endOfDay(col: Column, timeZoneId:String): Column = // equal to pass Some
```
the final code will be better to read

Reply:
@alfonsorr - Yea, I agree we should go with the method overloading approach, also because it's more consistent with the rest of the Spark API.

Think public Spark interfaces should avoid Optional values cause methods that don't have Option are easier to invoke via Java / Python (not sure about this reason, but think it's the case).

Reply:
In the absence of other concerns like consistency with Spark's API or Java compatibility, I think `Option` is the correct choice since it directly expresses our requirement. I don't see that the overloaded methods are easier to read than an `Option`.

I will change this to overloaded methods per your feedback, but I'll need to figure out how best to do it without duplicating the body of the method for each overloaded method.